test/src/main/java/org/vertexium/test/GraphTestBase.java

```diff
@@ -233,11 +236,47 @@ public void testAddStreamingPropertyValue() throws IOException, InterruptedExcep
         assertTrue(largePropertyName + " was " + propLargeValue.getClass().getName(), propLargeValue instanceof StreamingPropertyValue);
         value = (StreamingPropertyValue) propLargeValue;
         assertEquals(String.class, value.getValueType());
-        assertEquals(expectedLargeValue.getBytes().length, value.getLength());
+        assertEquals(expectedLargeValue.getBytes().length, (long) value.getLength());
         assertEquals(expectedLargeValue, IOUtils.toString(value.getInputStream()));
         assertEquals(expectedLargeValue, IOUtils.toString(value.getInputStream()));
     }
 
+    @Test
+    public void testStreamingPropertyDecreasingSize() throws IOException, InterruptedException {
+        Metadata metadata = new Metadata();
+        Long timestamp = System.currentTimeMillis();
+        String expectedValue = IOUtils.toString(new LargeStringInputStream(LARGE_PROPERTY_VALUE_SIZE));
+        PropertyValue propLarge = StreamingPropertyValue.create(
+                new ByteArrayInputStream(expectedValue.getBytes()),
+                String.class,
+                expectedValue.length()
+        );
+        graph.prepareVertex("v1", VISIBILITY_A)
+                .addPropertyValue("key1", "largeProp", propLarge, metadata, timestamp, VISIBILITY_A)
+                .save(AUTHORIZATIONS_A_AND_B);
+        graph.flush();
+
+        Vertex v1 = graph.getVertex("v1", AUTHORIZATIONS_A_AND_B);
+        StreamingPropertyValue spv = (StreamingPropertyValue) v1.getPropertyValue("key1", "largeProp");
+        assertEquals(expectedValue, spv.readToString());
+
+        // now save a smaller value, making sure it gets truncated
+        expectedValue = "small";
+        propLarge = StreamingPropertyValue.create(
+                new ByteArrayInputStream(expectedValue.getBytes()),
+                String.class,
+                expectedValue.length()
+        );
+        graph.prepareVertex("v1", VISIBILITY_A)
+                .addPropertyValue("key1", "largeProp", propLarge, metadata, timestamp + 1, VISIBILITY_A)
```

Comment:
My implementation without the metadata (length field) still worked because I only read the data from the property timestamp. But I added the metadata because it might be useful later.

Super weird but if I take this +1 off of timestamp I end up with duplicate columns (same key) here: https://github.com/visallo/vertexium/pull/195/files#diff-608da76fc5e28276fa9e63d6658f2af7R115 one column with the smaller value and one value with the larger value.